test/url_test.js

```diff
@@ -110,14 +110,27 @@ var url = require('url');
     });
 
     it('should validate period', function () {
-      assert.throws(function () {
-        new OTP({
-          secret: 'hello',
-          label: 'that',
-          period: 'hello',
-          counter: 0
-        }).url();
-      }, /invalid period `hello`/);
+      // period only matters for TOTP.  throw error when TOTP has invalid period
+      // When type is HOTP, invalid period is ignored
+      if (klass === 'TOTP') {
+        assert.throws(function () {
```

Comment:
I don't get it, please kindly elaborate. Which function do I need to define? Thanks.